src/main/kotlin/com/toasttab/pulseman/entities/ProjectSettingsV2.kt

```diff
@@ -15,20 +15,40 @@
 
 package com.toasttab.pulseman.entities
 
-import com.toasttab.pulseman.AppStrings.CONFIG_VERSION_NOT_SUPPORTED
+import com.toasttab.pulseman.state.protocol.protobuf.ProtobufTabValuesV3
+import com.toasttab.pulseman.state.protocol.text.TextTabValuesV3
 
+/**
+ * This is deprecated, keeping it for backwards compatibility with old config format, will eventually delete altogether.
+ */
+@Deprecated(
+    "This is an old save format, use ProjectSettingsV3 instead",
+    replaceWith = ReplaceWith("ProjectSettingsV3"),
+    level = DeprecationLevel.WARNING
+)
 data class ProjectSettingsV2(
     val configVersion: String,
-    val tabs: List<TabValuesV2>
-) {
-    init {
-        require(supportedConfigVersions.contains(configVersion)) {
-            "$configVersion $CONFIG_VERSION_NOT_SUPPORTED"
+    @Suppress("DEPRECATION") val tabs: List<TabValuesV2>
+) : ProjectSettings {
+    override fun toV3(): List<TabValuesV3> {
+        return tabs.map { tab ->
+            TabValuesV3(
+                tabName = tab.tabName,
+                topic = tab.topic,
+                serviceUrl = tab.serviceUrl,
+                selectedAuthClass = tab.selectedAuthClass,
+                authJsonParameters = tab.authJsonParameters,
+                propertyMap = tab.propertyMap,
+                serializationFormat = tab.serializationFormat,
+                protobufSettings = ProtobufTabValuesV3(
+                    code = tab.protobufSettings?.code,
+                    selectedClass = tab.protobufSettings?.selectedClassSend
```

Comment:
This was changed to only save a single selectedClass, the receive value has been discarded